agent/acl_endpoint_test.go

```diff
@@ -592,5 +645,251 @@ func TestACL_HTTP(t *testing.T) {
+			require.NoError(t, err)
+
+			token, ok := obj.(*structs.ACLToken)
+			require.True(t, ok)
+
+			// 36 = length of the string form of uuids
+			require.Equal(t, tokenInput.SecretID, token.SecretID)
+			require.Equal(t, tokenInput.AccessorID, token.AccessorID)
+			require.Equal(t, tokenInput.Description, token.Description)
+			require.Equal(t, tokenInput.Policies, token.Policies)
+			require.True(t, token.CreateIndex > 0)
+			require.Equal(t, token.CreateIndex, token.ModifyIndex)
+			require.NotNil(t, token.Hash)
+			require.NotEqual(t, token.Hash, []byte{})
+
+			idMap["token-test"] = token.AccessorID
+			tokenMap[token.AccessorID] = token
+		})
+
+		t.Run("Create with Accessor Dup", func(t *testing.T) {
+			tokenInput := &structs.ACLToken{
+				AccessorID:  "dee863fa-e548-4c61-a96f-9aa07999249f",
+				Description: "test",
+				Policies: []structs.ACLTokenPolicyLink{
+					structs.ACLTokenPolicyLink{
+						ID:   idMap["policy-test"],
+						Name: policyMap[idMap["policy-test"]].Name,
+					},
+					structs.ACLTokenPolicyLink{
+						ID:   idMap["policy-read-all-nodes"],
+						Name: policyMap[idMap["policy-read-all-nodes"]].Name,
+					},
+				},
+			}
+
+			req, _ := http.NewRequest("PUT", "/v1/acl/token?token=root", jsonBody(tokenInput))
+			resp := httptest.NewRecorder()
+			_, err := a.srv.ACLTokenCreate(resp, req)
+			require.Error(t, err)
+		})
+
+		t.Run("Create with Secret as Accessor Dup", func(t *testing.T) {
+			tokenInput := &structs.ACLToken{
+				SecretID:    "dee863fa-e548-4c61-a96f-9aa07999249f",
+				Description: "test",
+				Policies: []structs.ACLTokenPolicyLink{
+					structs.ACLTokenPolicyLink{
+						ID:   idMap["policy-test"],
+						Name: policyMap[idMap["policy-test"]].Name,
+					},
+					structs.ACLTokenPolicyLink{
+						ID:   idMap["policy-read-all-nodes"],
+						Name: policyMap[idMap["policy-read-all-nodes"]].Name,
+					},
+				},
+			}
+
+			req, _ := http.NewRequest("PUT", "/v1/acl/token?token=root", jsonBody(tokenInput))
+			resp := httptest.NewRecorder()
+			_, err := a.srv.ACLTokenCreate(resp, req)
+			require.Error(t, err)
+		})
+
+		t.Run("Create with Secret Dup", func(t *testing.T) {
+			tokenInput := &structs.ACLToken{
+				SecretID:    "10126ffa-b28f-4137-b9a9-e89ab1e97c5b",
+				Description: "test",
+				Policies: []structs.ACLTokenPolicyLink{
+					structs.ACLTokenPolicyLink{
+						ID:   idMap["policy-test"],
+						Name: policyMap[idMap["policy-test"]].Name,
+					},
+					structs.ACLTokenPolicyLink{
+						ID:   idMap["policy-read-all-nodes"],
+						Name: policyMap[idMap["policy-read-all-nodes"]].Name,
+					},
+				},
+			}
+
+			req, _ := http.NewRequest("PUT", "/v1/acl/token?token=root", jsonBody(tokenInput))
+			resp := httptest.NewRecorder()
+			_, err := a.srv.ACLTokenCreate(resp, req)
+			require.Error(t, err)
+		})
+
+		t.Run("Create with Accessor as Secret Dup", func(t *testing.T) {
+			tokenInput := &structs.ACLToken{
+				SecretID:    "10126ffa-b28f-4137-b9a9-e89ab1e97c5b",
```

Comment:
From the subtest description, I think you meant to specify `AccessorID` explicitly here.